src/components/Searchbar/SearchBox/SearchBox.tsx

```diff
@@ -31,185 +31,189 @@ type SearchBoxProps = {
-    const min = 20
-    const max = 80
-    const placeholderItems = Array.from({ length: 6 }, () => ({ id: undefined }))
-    return placeholderItems.map((_, idx) => {
-      const generatedWidth = Math.floor(Math.random() * (max - min)) + min
-      return (
-        <PlaceholderWrapper key={`placeholder-${idx}`}>
-          <SkeletonAvatar width="32px" height="32px" rounded />
-          <SkeletonLoader width={`${generatedWidth}%`} height="16px" />
-        </PlaceholderWrapper>
-      )
-    })
-  }, [])
+    const scrollToSelectedItem = useCallback(
+      (top: number, title?: string | null) => {
+        onSelectItem(title)
+        if (!containerRef.current) {
+          return
+        }
+        const { offsetHeight } = containerRef.current
+        if (selectedItem === 0 || top < offsetHeight) {
+          containerRef?.current?.scrollTo(0, 0)
+        }
+        if (top >= offsetHeight + (top < topRef.current ? -250 : -50)) {
+          containerRef?.current?.scrollTo(0, top + (top < topRef.current ? -50 : -250))
+        }
+        topRef.current = top
+      },
+      [onSelectItem, selectedItem]
+    )
 
-  const handleRecentSearchDelete = (title: string) => {
-    deleteRecentSearch(title)
-  }
+    const generatePlaceholders = useMemo(() => {
+      const min = 20
+      const max = 80
+      const placeholderItems = Array.from({ length: 6 }, () => ({ id: undefined }))
+      return placeholderItems.map((_, idx) => {
+        const generatedWidth = Math.floor(Math.random() * (max - min)) + min
+        return (
+          <PlaceholderWrapper key={`placeholder-${idx}`}>
+            <SkeletonAvatar width="32px" height="32px" rounded />
+            <SkeletonLoader width={`${generatedWidth}%`} height="16px" />
+          </PlaceholderWrapper>
+        )
+      })
+    }, [])
+
+    const handleRecentSearchDelete = (title: string) => {
+      deleteRecentSearch(title)
+    }
 
-  const filteredRecentSearches = searchQuery.length
-    ? recentSearches
-        .filter((item) =>
-          new RegExp(`${searchQuery.replace(SPECIAL_CHARACTERS, '\\$&').replace(/\s+/g, '|')}`, 'i').test(
-            item.title || ''
+    const filteredRecentSearches = searchQuery.length
+      ? recentSearches
+          .filter((item) =>
+            new RegExp(`${searchQuery.replace(SPECIAL_CHARACTERS, '\\$&').replace(/\s+/g, '|')}`, 'i').test(
+              item.title || ''
+            )
           )
-        )
-        .slice(0, 3)
-    : recentSearches
-  const slicedVideos = videos.slice(0, 3)
-  const slicedChannels = channels.slice(0, 3)
+          .slice(0, 3)
+      : recentSearches
+    const slicedVideos = videos.slice(0, 3)
+    const slicedChannels = channels.slice(0, 3)
 
-  useEffect(() => {
-    handleSetNumberOfItems(filteredRecentSearches.length + slicedVideos.length + slicedChannels.length)
-  }, [handleSetNumberOfItems, filteredRecentSearches.length, slicedVideos.length, slicedChannels.length])
+    useEffect(() => {
```

Comment:
I know this might be a simple one but it's always good to have a small comment on top of the useEffects to clarify what they do